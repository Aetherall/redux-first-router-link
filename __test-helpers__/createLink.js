import React from 'react'
import renderer from 'react-test-renderer'
import { createStore, applyMiddleware } from 'redux'
import { Provider } from 'react-redux'
import createHistory from 'history/createMemoryHistory'
import { connectTypes } from 'pure-redux-router'

import Link from '../src/Link'


export default (props) => {
  const routesMap = {
    FIRST: '/first',
    SECOND: '/second/:param',
  }

  const history = createHistory({
    initialEntries: ['/'],
    initialIndex: 0,
    keyLength: 6,
  })

  const { middleware, reducer: locationReducer } = connectTypes(history, routesMap)

  const middlewares = applyMiddleware(middleware)

  const reducer = (state = {}, action = {}) => ({
    location: locationReducer(state.location, action),
  })

  const store = createStore(reducer, undefined, middlewares)

  const link = <Link {...props} />
  const component = renderer.create(<Provider store={store}>{link}</Provider>)

  return {
    tree: component.toJSON(),
    store,
  }
}


export const event = { preventDefault: () => undefined }