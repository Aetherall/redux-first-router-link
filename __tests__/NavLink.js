import { NOT_FOUND } from 'redux-first-router'
import { createNavLink, event } from '../__test-helpers__/createLink'

test('NON-EXACT: show active class', () => {
  const { tree } = createNavLink('/first', {
    to: '/first',
    activeClassName: 'active'
  }) /*? $.tree */

  expect(tree).toMatchSnapshot() /*? store.getState() */
})

test('EXACT: DONT show active class', () => {
  const { tree } = createNavLink('/second/dog', {
    to: '/second',
    exact: true,
    activeClassName: 'active'
  }) /*? $.tree */

  expect(tree).toMatchSnapshot() /*? store.getState() */
})

test('STRICT: DONT show active class', () => {
  const { tree } = createNavLink('/first', {
    to: '/first/',
    strict: true,
    activeClassName: 'active'
  }) /*? $.tree */

  expect(tree).toMatchSnapshot() /*? store.getState() */
})

test('show activeStyle', () => {
  const { tree } = createNavLink('/first', {
    to: '/first',
    activeStyle: { color: 'red' }
  }) /*? $.tree */

  expect(tree).toMatchSnapshot() /*? store.getState() */
})

test('combine with existing styles and class', () => {
  const { tree } = createNavLink('/first', {
    to: '/first',
    className: 'foo',
    style: { fontSize: 32 },
    activeClassName: 'active',
    activeStyle: { color: 'red' }
  }) /*? $.tree */

  expect(tree).toMatchSnapshot() /*? store.getState() */
})

test('reacts to state changes (onClick)', () => {
  const { tree, store, component } = createNavLink('/first', {
    to: '/second/bar',
    activeClassName: 'active'
  })

  expect(tree.props.className).not.toBeDefined()
  expect(tree).toMatchSnapshot() /*? tree */

  // store.dispatch({ type: 'SECOND', payload: { param: 'bar' } })
  tree.props.onClick(event) // this dispatches above action (obviously)

  const tree2 = component.toJSON() /*? */

  expect(tree2.props.className).toEqual('active')
  expect(tree2).toMatchSnapshot()
})

test('isActive returns true', () => {
  const { tree, store } = createNavLink('/first', {
    to: '/first',
    activeClassName: 'active',
    isActive: (match, location) => {
      expect(match).toMatchSnapshot()
      expect(location).toMatchSnapshot()
      return true
    }
  })

  expect(tree).toMatchSnapshot()
})

test('isActive return false', () => {
  const { tree, store } = createNavLink('/first', {
    to: '/first',
    activeClassName: 'active',
    isActive: (match, location) => {
      expect(match).toMatchSnapshot()
      expect(location).toMatchSnapshot()
      return false
    }
  })

  expect(tree).toMatchSnapshot()
})
