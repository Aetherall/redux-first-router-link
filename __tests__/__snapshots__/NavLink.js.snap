// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EXACT: DONT show active class 1`] = `
<a
  className={undefined}
  href="/second"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`NON-EXACT: show active class 1`] = `
<a
  className="active"
  href="/first"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`STRICT: DONT show active class 1`] = `
<a
  className={undefined}
  href="/first/"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`combine with existing styles and class 1`] = `
<a
  className="foo active"
  href="/first"
  onClick={[Function]}
  style={
    Object {
      "color": "red",
      "fontSize": 32,
    }
  }
/>
`;

exports[`isActive return false 1`] = `
Object {
  "isExact": true,
  "params": Object {},
  "path": "/first",
  "url": "/first",
}
`;

exports[`isActive return false 2`] = `
Object {
  "hasSSR": undefined,
  "history": undefined,
  "kind": "load",
  "pathname": "/first",
  "payload": Object {},
  "prev": Object {
    "pathname": "",
    "payload": Object {},
    "type": "",
  },
  "routesMap": Object {
    "FIRST": "/first",
    "SECOND": "/second/:param",
  },
  "type": "FIRST",
}
`;

exports[`isActive return false 3`] = `
<a
  className={undefined}
  href="/first"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`isActive returns true 1`] = `
Object {
  "isExact": true,
  "params": Object {},
  "path": "/first",
  "url": "/first",
}
`;

exports[`isActive returns true 2`] = `
Object {
  "hasSSR": undefined,
  "history": undefined,
  "kind": "load",
  "pathname": "/first",
  "payload": Object {},
  "prev": Object {
    "pathname": "",
    "payload": Object {},
    "type": "",
  },
  "routesMap": Object {
    "FIRST": "/first",
    "SECOND": "/second/:param",
  },
  "type": "FIRST",
}
`;

exports[`isActive returns true 3`] = `
<a
  className="active"
  href="/first"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`reacts to state changes (onClick) 1`] = `
<a
  className={undefined}
  href="/second/bar"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`reacts to state changes (onClick) 2`] = `
<a
  className="active"
  href="/second/bar"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`show activeStyle 1`] = `
<a
  className=""
  href="/first"
  onClick={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
/>
`;
